lib/plugins/aws/package/compile/events/cloudFront/index.js

```diff
@@ -85,13 +85,45 @@ class AwsCompileCloudFrontEvents {
       properties: {
         AllowedMethods: {
           oneOf: [
-            { enum: ['GET', 'HEAD'] },
-            { enum: ['GET', 'HEAD', 'OPTIONS'] },
-            { enum: ['GET', 'HEAD', 'OPTIONS', 'PUT', 'PATCH', 'POST', 'DELETE'] },
+            {
+              uniqueItems: true,
+              minItems: 2,
+              items: { enum: ['GET', 'HEAD'] },
+            },
+            {
+              uniqueItems: true,
+              minItems: 3,
+              items: { enum: ['GET', 'HEAD', 'OPTIONS'] },
+            },
+            {
+              uniqueItems: true,
+              minItems: 7,
+              items: { enum: ['GET', 'HEAD', 'OPTIONS', 'PUT', 'PATCH', 'POST', 'DELETE'] },
+            },
           ],
         },
         CachedMethods: {
-          oneOf: [{ enum: ['GET', 'HEAD'] }, { enum: ['GET', 'HEAD', 'OPTIONS'] }],
+          oneOf: [
+            {
+              uniqueItems: true,
+              minItems: 2,
+              items: { enum: ['GET', 'HEAD'] },
+            },
+            {
+              uniqueItems: true,
+              minItems: 3,
+              items: { enum: ['GET', 'HEAD', 'OPTIONS'] },
+            },
+          ],
+        },
+        ForwardedValues: {
+          Cookies: {
+            Forward: { type: 'string' },
+            WhitelistedNames: { items: { type: 'string' } },
+          },
+          Headers: { items: { type: 'string' } },
+          QueryString: { type: 'boolean' },
+          QueryStringCacheKeys: { items: { type: 'string' } },
```

Comment:
We've initially left `ForwardedValues` out, as it's hardcoded: https://github.com/serverless/serverless/blob/eb5e54847e6e2f6b89a1b5325df4d8421efe479a/lib/plugins/aws/package/compile/events/cloudFront/index.js#L294-L296

Still, indeed it can be overriden.

Anyway, this schema needs to be fixed, in current form it's ineffective (all content is ignored by AJV. unfortunately it doesn't report on unrecognized content in schema)

Proper form for object definitions is as:


```javascript
ForwardedValues: {
  type: "object",
  properties: {
    ...
  },
  additionalProperties: false,
  required: [...]
}
```

Reply:
Thanks! Fixed! Yeah, it would be easier if AJV warned :)